specs/phase1/beacon-chain.md

```diff
@@ -12,150 +12,115 @@
-        - [`apply_shard_transition`](#apply_shard_transition)
-        - [`process_crosslink_for_shard`](#process_crosslink_for_shard)
-        - [`process_crosslinks`](#process_crosslinks)
-        - [`verify_empty_shard_transition`](#verify_empty_shard_transition)
-        - [`process_shard_transitions`](#process_shard_transitions)
-      - [New default validator for deposits](#new-default-validator-for-deposits)
-    - [Light client processing](#light-client-processing)
+  - [New Attestation processing](#new-attestation-processing)
+    - [Updated `process_attestation`](#updated-process_attestation)
+    - [`update_pending_votes`](#update_pending_votes)
+    - [`process_shard_header`](#process_shard_header)
+  - [Shard transition processing](#shard-transition-processing)
   - [Epoch transition](#epoch-transition)
+    - [Pending headers](#pending-headers)
     - [Phase 1 final updates](#phase-1-final-updates)
     - [Custody game updates](#custody-game-updates)
-    - [Online-tracking](#online-tracking)
-    - [Light client committee updates](#light-client-committee-updates)
 
 <!-- END doctoc generated TOC please keep comment here to allow auto update -->
 
 ## Introduction
 
-This document describes the extensions made to the Phase 0 design of The Beacon Chain
- to facilitate the new shards as part of Phase 1 of Eth2.
+This document describes the extensions made to the Phase 0 design of The Beacon Chain to support data sharding, based on the ideas [here](https://hackmd.io/@HWeNw8hNRimMm2m2GH56Cw/r1XzqYIOv) and more broadly [here](https://arxiv.org/abs/1809.09044), using Kate commitments to commit to data to remove any need for fraud proofs (and hence, safety-critical synchrony assumptions) in the design.
 
 ## Custom types
 
 We define the following Python custom types for type hinting and readability:
 
 | Name | SSZ equivalent | Description |
 | - | - | - |
-| `Shard` | `uint64` | a shard number |
-| `OnlineEpochs` | `uint8` | online countdown epochs |
+| `Shard` | `uint64` | A shard number |
+| `BLSCommitment` | `bytes48` | A G1 curve point |
 
 ## Configuration
 
-Configuration is not namespaced. Instead it is strictly an extension;
- no constants of phase 0 change, but new constants are adopted for changing behaviors.
-
 ### Misc
 
-| Name | Value |
-| - | - | 
-| `MAX_SHARDS` | `uint64(2**10)` (= 1024) |
-| `INITIAL_ACTIVE_SHARDS` | `uint64(2**6)` (= 64) |
-| `LIGHT_CLIENT_COMMITTEE_SIZE` | `uint64(2**7)` (= 128) |
-| `GASPRICE_ADJUSTMENT_COEFFICIENT` | `uint64(2**3)` (= 8) | 
+| Name | Value | Notes |
+| - | - | - |
+| `MAX_SHARDS` | `uint64(2**10)` (= 1024) | Theoretical max shard count (used to determine data structure sizes) |
+| `INITIAL_ACTIVE_SHARDS` | `uint64(2**6)` (= 64) | Initial shard count |
+| `GASPRICE_ADJUSTMENT_COEFFICIENT` | `uint64(2**3)` (= 8) | Gasprice may decrease/increase by at most exp(1 / this value) *per epoch* |
+| `MAX_SHARD_HEADERS_PER_SHARD` | `4` | |
```

Comment:
Where does this value come from? Is this correlated with another part of the system?

Reply:
We need at least 2 per shard to cover the normal case (the header proposed by the proposer, and the null header); 4 leaves some room to handle malicious multi-proposals as well. No other reasoning than that.

Reply:
I think one annoying property is that a single shard proposer could fill all the proposals; thus an attacker only has to equivocate with one validator to fill up all proposals for one epoch.
Would it make sense to put a per-shard and per-height limit on this as well?